Docs/source/building/summit.rst

```diff
@@ -3,87 +3,106 @@
 
-.. note::
+   # required dependencies
+   module load gcc/6.4.0
+   module load cuda
 
-   To compile and run WarpX on Summit CPUs, the cuda module is not necessary.
+   # optional: faster re-builds
+   module load ccache
 
-   But to build with the spectral solver, PSATD, you need to use an MPI-enabled version of FFTW like the `fftw` module (because the `cuFFT` tool is not available when cuda is not loaded).
+   # optional: for PSATD support
+   module load fftw
 
-   ::
+   # optional: for QED support
+   module load boost/1.66.0
 
-      module load gcc
-      module load fftw
-      make -j 16 COMP=gcc USE_PSATD=TRUE
+   # optional: for openPMD support
+   module load cmake
+   module load hdf5/1.10.4
+   module load adios2/2.5.0
+   export PKG_CONFIG_PATH=$HOME/sw/openPMD-api-install/lib64/pkgconfig:$PKG_CONFIG_PATH
+   export CMAKE_PREFIX_PATH=$HOME/sw/openPMD-api-install:$CMAKE_PREFIX_PATH
 
+   # optional: just an additional text editor
+   module load nano
 
+   # optional: an alias to request an interactive node for two hours
+   alias getNode="bsub -P $proj -W 2:00 -nnodes 1 -Is /bin/bash"
 
+   # fix system defaults: do not escape $ with a \ on tab completion
+   shopt -s direxpand
 
-.. _building-summit-openPMD:
+   # compiler environment hints
+   export CC=$(which gcc)
+   export CXX=$(which g++)
+   export CUDACXX=$(which nvcc)
+   export CUDAHOSTCXX=$(which g++)
 
-Building WarpX with openPMD support
------------------------------------
 
-First, load the appropriate modules:
+We recommend to store the above lines in a file, such as ``$HOME/warpx.profile``, and load it into your shell after a login:
 
 .. code-block:: bash
 
-    module load gcc
-    module load cuda
-    module load cmake
-    module load hdf5/1.10.4
-    module load adios2/2.5.0
+   source $HOME/warpx.profile
 
-    export CC=$(which gcc)
-    export CXX=$(which g++)
-    export CUDACXX=$(which nvcc)
-    export CUDAHOSTCXX=$(which g++)
+Use the following commands to download the WarpX source code and switch to the correct branch:
+
+.. code-block:: bash
 
-Then, in the ``warpx_directory``, download and build openPMD-api:
+   mkdir ~/src
+   cd ~/src
+
+   git clone https://github.com/ECP-WarpX/WarpX.git warpx
+   git clone --branch QED https://bitbucket.org/berkeleylab/picsar.git
+   git clone --branch development https://github.com/AMReX-Codes/amrex.git
+
+Optionally, download and build openPMD-api for I/O:
 
 .. code-block:: bash
 
    git clone https://github.com/openPMD/openPMD-api.git
    mkdir openPMD-api-build
    cd openPMD-api-build
-   cmake ../openPMD-api -DopenPMD_USE_PYTHON=OFF -DCMAKE_INSTALL_PREFIX=../openPMD-install/ -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON -DCMAKE_INSTALL_RPATH='$ORIGIN' -DMPIEXEC_EXECUTABLE=$(which jsrun)
+   cmake ../openPMD-api -DopenPMD_USE_PYTHON=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/openPMD-api-install/ -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON -DCMAKE_INSTALL_RPATH='$ORIGIN' -DMPIEXEC_EXECUTABLE=$(which jsrun)
```

Comment:
I have observed an error of type `device or resource busy` during this line, which does not does not occur all the time. Not sure if this is a temporary issue with Summit or a possible problem with openpmd.

Reply:
Thank you for the note, that's interesting.
I think this might occur temporarily when several users are overloading the filesystem of the head-node. If this happens more often we can add a bit more involved build in tmpfs or I provide the software permanently in `$WORLDWORK/aph114/`.